libp2p/protocols/pubsub/pubsub.nim

```diff
@@ -87,11 +105,14 @@ proc sendSubs*(p: PubSub,
 
     await peer.sendSubOpts(topics, subscribe)
   except CancelledError as exc:
-    p.handleDisconnect(peer)
+    if not(isNil(peer)) and not(isNil(peer.conn)):
```

Comment:
if peer is `nil` something is broken tho isn't it? Thought we fixed those issues

Reply:
this is just a bit of defencive programming here.